pkg/audit/reporter.go

```diff
@@ -106,6 +113,13 @@ func (reporter *Reporter) recordAuditFailStatus(ctx context.Context, failedAudit
 			failed = append(failed, nodeID)
 			errlist.Add(err)
 		}
+
+		// TODO(kaloyan): Perhaps, this should be executed in the same Tx as overlay.UpdateStats above
+		_, err = reporter.containment.Delete(ctx, nodeID)
```

Comment:
I don't think we should delete from containment here. A node might fail a different audit than the one it was contained for, but we should still audit it for the share it was contained for (unless I'm confused)